### PR TITLE
Various tweaks to reserved cluster state tests

### PR DESCRIPTION
**Pull request by thecoop**
Use more modern test assertions for reserved cluster state tests

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/reservedstate/action/ReservedClusterSettingsActionTests.java [hunk 1]
```diff
@@ -142,6 +143,6 @@ public void testSettingNameNormalization() throws Exception {
             """;
         TransformState newState2 = processJSON(testAction, prevState, jsonRemoval);
         assertThat(newState2.keys(), containsInAnyOrder("dummy.setting2"));
-        assertThat(newState2.state().metadata().persistentSettings().get("dummy.setting2"), equalTo("value2"));
+        assertThat(newState2.state().metadata().persistentSettings().get("dummy.setting2"), is("value2"));
```
Comment (prdoyle) [inline server/src/test/java/org/elasticsearch/reservedstate/action/ReservedClusterSettingsActionTests.java:146]: Why is this preferred?

File: server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java [hunk 2]
```diff
@@ -367,9 +334,8 @@ public void onFailure(Exception e) {}
         );
 
         var trialRunResult = controller.trialRun("namespace_one", state, chunk, new LinkedHashSet<>(orderedHandlers));
-        assertEquals(0, trialRunResult.nonStateTransforms().size());
-        assertEquals(1, trialRunResult.errors().size());
-        assertTrue(trialRunResult.errors().get(0).contains("Error processing one state change:"));
+        assertThat(trialRunResult.nonStateTransforms(), empty());
+        assertThat(trialRunResult.errors(), contains(containsString("Error processing one state change:")));
```
Comment (prdoyle) [inline server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java:338]: Does this lose the assertion that `trialRunResult.errors()` contains just one entry?

File: server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java [hunk 3]
```diff
@@ -506,8 +473,8 @@ public void testCheckAndReportError() {
 
         var version = new ReservedStateVersion(2L, Version.CURRENT);
         var error = controller.checkAndReportError("test", List.of("test error"), version);
-        assertThat(error, allOf(notNullValue(), instanceOf(IllegalStateException.class)));
-        assertEquals("Error processing state change request for test, errors: test error", error.getMessage());
+        assertThat(error, instanceOf(IllegalStateException.class));
+        assertThat(error.getMessage(), is("Error processing state change request for test, errors: test error"));
```
Comment (prdoyle) [inline server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java:477]: Again I wonder why this is preferred? For uniformity with the other asserts in the same test?

**Review by prdoyle — APPROVED** [review]
LGTM. A ton of little improvements in there.

File: server/src/test/java/org/elasticsearch/reservedstate/action/ReservedClusterSettingsActionTests.java (hunk 1 above)
Comment (thecoop) [reply server/src/test/java/org/elasticsearch/reservedstate/action/ReservedClusterSettingsActionTests.java:146]: To match the others (where I've also used `is`)

File: server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java (hunk 3 above)
Comment (thecoop) [reply server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java:477]: Exactly

**Pull request merged**